packages/test/test-end-to-end-tests/.eslintrc.cjs

```diff
@@ -61,6 +62,20 @@ module.exports = {
 			@fluid-private/test-end-to-end-tests:     at ExportMap.get (/home/tylerbu/code/release-1/node_modules/.pnpm/eslint-plugin-i@2.29.0_j7h7oj6rrhtikhzta4fgkou42e/node_modules/eslint-plugin-i/lib/ExportMap.js:792:465)
 		 */
 		"import/no-deprecated": "off",
+		"regex/invalid-error": [
+			"error",
+			[
+				{
+					id: "invalidDescribeCompat",
+					message:
+						"Invalid describeCompat usage. The inner describeCompat block should have have arguments defined. Using the TestObjectProvider from the outer block inside the inner block may lead to unexpected behavior.",
+					regex: `.*describeCompat\\(".*?",\\s+".*?",\\s+\\(.*?\\)\\s+=>\\s+\\{\\s+describeCompat\\(".*?",\\s+".*?",\\s+\\(\\)\\s+=>\\s+\\{.*`,
```

Comment:
Did we consider adding a custom rule like [the others we already have](https://github.com/microsoft/FluidFramework/tree/main/common/build/eslint-plugin-fluid/src/rules)? Troubleshooting and ensuring all cases are handled with regex feels tricky, to say the least. Using `context.sourceCode` provided by ESLint to custom rules and its `getAncestors()` method, it seems like we could do quick work of this check ("is the current AST node a call to ` describeCompat` and the ancestor node also a call to `describeCompat`? error out").

Reply:
What kind of alternative were explored here? Is there an existing runtime check? Is that an option that was explored? What do we see as the benefits for a lint rule over other types of validation

Reply:
I think you can implement a runtime check using `before` and `after` hooks paired with some module-level state in `describeCompat.ts`. Depending on the guarantees that mocha provides around test execution, the module-level state might be as simple as a single 'inUse' variable, but you could make this more robust to things like parallel test execution by keying the different suites' "inUse" status on something like suite name.